ehrapy/preprocessing/_bias.py

```diff
@@ -0,0 +1,119 @@
+        corr_threshold: The threshold for the correlation coefficient between two features to be considered of interest. Defaults to 0.5.
+        smd_threshold: The threshold for the standardized mean difference between two features to be considered of interest. Defaults to 0.5.
+        feature_importance_threshold: The threshold for the feature importance of a sensitive feature for predicting another feature to be considered
+            of interest. Defaults to 0.1.
+        prediction_confidence_threshold: The threshold for the prediction confidence (R2 or accuracy) of a sensitive feature for predicting another
+            feature to be considered of interest. Defaults to 0.5.
+        corr_method: The correlation method to use. Choose between "pearson" and "spearman". Defaults to "spearman".
+    """
+    from ehrapy.tools import rank_features_supervised
+
+    if sensitive_features == "all":
+        sensitive_features = adata.var_names
+
+    correlations = _feature_correlations(adata, method=corr_method)
+    adata.varp["correlation"] = correlations
+
+    for feature in sensitive_features:
+        for comp_feature in adata.var_names:
+            if correlations.loc[feature, comp_feature] > corr_threshold:
+                logg.warning(
+                    f"Feature {feature} is highly correlated with {comp_feature} (correlation coefficient ≈{correlations.loc[feature, comp_feature]:.3f})."
+                )  # TODO: How do we print results?
+
+    smd_dict = _standardized_mean_differences(adata, sensitive_features)
+    for feature in sensitive_features:
+        abs_smd = smd_dict[feature].abs()
+        for comp_feature in adata.var_names:
+            if abs_smd[comp_feature].max() > smd_threshold:
+                logg.warning(
+                    f"Feature {comp_feature} has a high standardized mean difference with {feature}."
+                )  # TODO: Do we look at / print groups individually?
+
+    for prediction_feature in adata.var_names:
+        prediction_score = rank_features_supervised(
+            adata,
+            prediction_feature,
+            input_features="all",
+            model="rf",
+            key_added=f"{prediction_feature}_feature_importances",
+            percent_output=True,
+            logging=False,
+            return_score=True,
+        )
+        for feature in sensitive_features:
+            feature_importance = adata.var[f"{prediction_feature}_feature_importances"][feature] / 100
+            if feature_importance > feature_importance_threshold and prediction_score > prediction_confidence_threshold:
+                logg.warning(
+                    f"Feature {feature} has a high feature importance for predicting {prediction_feature} (importance in %: {feature_importance:.3f}, prediction score: {prediction_score:.3f})."
+                )
+
+
+def _feature_correlations(adata: AnnData, method: Literal["pearson", "spearman"] = "spearman"):
```

Comment:
Misses return type

Reply:
Generally, I've learn that it's fine to keep the docstrings more minimal for such simple functions. If the function name is self explanatory (which it should be here!) we can even omit the whole docstring.

But in any case, you might even integrate this into the bigger function anyways.

Reply:
Yeah, I'll integrate it into the bigger function now so I'll delete this code section.